torch/nn/parallel/_functions.py

```diff
@@ -90,15 +101,16 @@ def forward(ctx, target_gpus, chunk_sizes, dim, input):
         ctx.dim = dim
         ctx.input_device = input.get_device() if input.device.type != "cpu" else -1
         streams = None
-        if torch.cuda.is_available() and ctx.input_device == -1:
+        ddp_gpu = _get_device_impl()[1]
```

Comment:
hmm, there are users using DDP for CPU-only training. Will that break after this change?

Reply:
> I guess not since this change keeps the original logic while might just check a different type of gpu?

Yes. This path is only for gpu, not related to cpu.